tests/operator_test.go

```diff
@@ -2081,7 +2083,22 @@ spec:
 				secret, err := coreClient.Secrets("openshift-monitoring").Get(context.Background(), "prometheus-k8s", metav1.GetOptions{})
 				Expect(err).ToNot(HaveOccurred())
 
-				data := secret.Data["prometheus.yaml"]
+				data, ok := secret.Data["prometheus.yaml"]
+				// In new versions of prometheus-operator, the configuration file is compressed in the secret
+				if !ok {
+					data, ok = secret.Data["prometheus.yaml.gz"]
+					Expect(ok).To(BeTrue())
+
+					By("Decompressing Prometheus' configuration data")
+					gzreader, err := gzip.NewReader(bytes.NewReader(data))
+					Expect(err).ToNot(HaveOccurred())
+
+					decompressed, err := ioutil.ReadAll(gzreader)
+					Expect(err).ToNot(HaveOccurred())
+
+					data = decompressed
```

Comment:
It looks good to me ,
I would create a function for the decomposition but it's not written anywhere 
else in the code so it's looks good like that as well.